contracts/contracts/SimpleWriteAccessController/library.cairo

```diff
@@ -0,0 +1,142 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+from ownable import Ownable_only_owner, Ownable_initializer
+
+struct Phase:
+    member id : felt
+    member aggregator_address : felt
+end
+
+@event
+func AddedAccess(user : felt):
+end
+
+@event
+func RemovedAccess(user : felt):
+end
+
+@event
+func CheckAccessEnabled():
+end
+
+@event
+func CheckAccessDisabled():
+end
+
+@storage_var
+func s_check_enabled() -> (checkEnabled : felt):
+end
+
+@storage_var
+func s_access_list(address : felt) -> (bool : felt):
+end
+
+# Adds an address to the access list
+@external
+func add_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    Ownable_only_owner()
+    _add_access(user)
+    return ()
+end
+
+func _add_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    let (has_access) = s_access_list.read(user)
+    if has_access == FALSE:
+        s_access_list.write(user, TRUE)
+        AddedAccess.emit(user)
+        return ()
+    end
+
+    return ()
+end
+
+# # Removes an address from the access list
+@external
+func remove_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(user : felt):
+    Ownable_only_owner()
+    let (has_access) = s_access_list.read(user)
+    if has_access == TRUE:
+        s_access_list.write(user, FALSE)
+        RemovedAccess.emit(user)
+        return ()
+    end
+
+    return ()
+end
+
+# makes the access check enforced
+@external
+func enable_access_check{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    Ownable_only_owner()
+    let (check_enabled) = s_check_enabled.read()
+    if check_enabled == FALSE:
+        s_check_enabled.write(TRUE)
+        CheckAccessEnabled.emit()
+        return ()
+    end
+
+    return ()
+end
+
+# makes the access check unenforced
+@external
+func disable_access_check{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    Ownable_only_owner()
+    let (check_enabled) = s_check_enabled.read()
+    if check_enabled == TRUE:
+        s_check_enabled.write(FALSE)
+        CheckAccessDisabled.emit()
+        return ()
+    end
+
+    return ()
+end
+
+namespace simple_write_access_controller:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        owner_address : felt
+    ):
+        Ownable_initializer(owner_address)
+        s_check_enabled.write(TRUE)
+
+        return ()
+    end
+
+    # Virtual method needs wrapper
+    func has_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        user : felt, data_len : felt, data : felt*
```

Comment:
I wonder if we should remove `data` from here since on Cairo we can't pass `calldata` through in a generic way anyway and it's not checked anywhere

Reply:
It's part of an interface that is unused now but could be used in the future by a non-simple access controller, for example, to verify a signature or something like that.

I'd suggest we leave it to keep our interfaces consistent.

Reply:
@krebernisak , @archseer about this comment, and the same below [contracts/contracts/SimpleWriteAccessController/library.cairo], resolved? Not sure why we have the same comment on same file